UIforETW/Utility.cpp

```diff
@@ -127,71 +460,126 @@ std::wstring ConvertToCRLF(const std::wstring& input)
 		if (c != '\r')
 			result += c;
 	}
-
 	return result;
 }
 
-void SetRegistryDWORD(HKEY root, const std::wstring& subkey, const std::wstring& valueName, DWORD value)
+void SetRegistryDWORD(const HKEY root, const std::wstring& subkey, const std::wstring& valueName, const DWORD value)
 {
 	HKEY key;
-	LONG result = RegOpenKeyEx(root, subkey.c_str(), 0, KEY_ALL_ACCESS, &key);
-	if (result == ERROR_SUCCESS)
+	if (!OpenRegKey(&key, root, subkey.c_str()))
+		return;
+
+	const LONG setResult = ::RegSetValueExW(key, valueName.c_str(), 0, REG_DWORD, reinterpret_cast<const BYTE*>(&value), sizeof(value));
+	if (setResult != ERROR_SUCCESS)
 	{
-		RegSetValueEx(key, valueName.c_str(), 0, REG_DWORD, reinterpret_cast<const BYTE*>(&value), sizeof(value));
-		RegCloseKey(key);
+		outputPrintf(L"Failed to set registry key `%s`.\n", subkey.c_str());

```

Comment:
Here, I think the real issue is that if `RegSetValueEx` fails, we might not be able to set `DisablePagingExecutive`, and then xperf/ETW stack walking wouldn't work. Since ETW is vastly less useful without stack walking, UIforETW is then also much less useful. 

In that case, we'd probably be better off handling that in `CUIforETWDlg::DisablePagingExecutive` instead. Otherwise I can `UIETWASSERT` this instead.